### PR TITLE
feat(eks-v2-alpha): use L1 CfnCluster to replace custom resource for EKS Cluster

### PR DESCRIPTION
**Pull request by xazhao**
### Reason for this change

Get rid of the custom resource to provision EKS cluster by using L1 CfnCluster.

### Description of changes

Replace custom resource with native L1 CfnCluster

### Description of how you validated changes
unit tests and integration tests

### Checklist
- [ ] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts [hunk 1]
```diff
@@ -1655,34 +1596,26 @@ export class Cluster extends ClusterBase {
...
-    }
-
     if (props.serviceIpv4Cidr && props.ipFamily == IpFamily.IP_V6) {
       throw new Error('Cannot specify serviceIpv4Cidr with ipFamily equal to IpFamily.IP_V6');
     }
 
     this.authenticationMode = props.authenticationMode;
 
-    const resource = this._clusterResource = new ClusterResource(this, 'Resource', {
+    const resource = this._clusterResource = new CfnCluster(this, 'Resource', {
```
Comment (pahud) [inline packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts:1605]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The Kubernetes API server for the Amazon EKS cluster is exposed if `endpointAccess` is set to `EndpointAccess.PUBLIC` in the `Cluster` resource or `endpointPublicAccess` is `true` in the `CfnCluster` resource. This configuration allows unauthorized internet access, increasing security risks such as data breaches and malicious activities. To secure the cluster, set `endpointAccess` to `EndpointAccess.PRIVATE` in the `Cluster` resource and `endpointPublicAccess` to `false` in the `CfnCluster` resource. Implement network access controls to restrict access to trusted sources within the AWS VPC. For more information refer to the documentation https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-eks.EndpointAccess.html.

**Review by iliapolo — CHANGES_REQUESTED** [review]
I'm seeing some files I didn't expect changing, but I can't see the change because the PR is too big. Lets talk about this.

File: packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts [hunk 2]
```diff
@@ -1758,15 +1681,18 @@ export class Cluster extends ClusterBase {
     // and configured to allow connections from itself.
     this.kubectlSecurityGroup = this.clusterSecurityGroup;
 
-    this.adminRole.assumeRolePolicy?.addStatements(new iam.PolicyStatement({
-      actions: ['sts:AssumeRole'],
-      principals: [this.kubectlLambdaRole],
-    }));
+    this.adminRole = new iam.Role(this, 'kubectlRole', {
+      assumedBy: new iam.CompositePrincipal(this.kubectlLambdaRole),
```
Comment (GavinZZ) [inline packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts:1685]: For my understanding, why do we need to use `CompositePrincipal` here?

File: packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts [hunk 3]
```diff
@@ -1535,7 +1480,7 @@ export class Cluster extends ClusterBase {
    * that manages it. If this cluster is not kubectl-enabled (i.e. uses the
    * stock `CfnCluster`), this is `undefined`.
    */
-  private readonly _clusterResource: ClusterResource;
+  private readonly _clusterResource: CfnCluster;
```
Comment (GavinZZ) [inline packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts:1483]: I think we should update the command related to this property.

File: packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts [hunk 4]
```diff
@@ -1545,7 +1490,7 @@ export class Cluster extends ClusterBase {
 
   private readonly version: KubernetesVersion;
 
-  private readonly logging?: { [key: string]: [ { [key: string]: any } ] };
+  private readonly logging?: { [key: string]: any };
```
Comment (GavinZZ) [inline packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts:1493]: nit: I think this can still be `{ [key: string]: { [key:string]: any} };`

**Review by GavinZZ — COMMENTED** [review]
Generally looks good as most of the changes are tests changes and files related to custom resource removal. A few questions:
1. In the design document, we talked about updating `fromKubectlProviderAttributes`. I don't see the change here, is this intended for a separate PR?
2. We also talked about updating `ICluster` interface to remove the unused properties but I don't see that here either.

**Comment by xazhao** [comment]
@GavinZZ Thanks for reviewing. Yes those changes will be in separate PRs.

File: packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts (hunk 2 above)
Comment (xazhao) [reply packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts:1685]: Actually no need to use `CompositePrincipal` here. Will update.

File: packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts (hunk 3 above)
Comment (xazhao) [reply packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts:1483]: Do you mean the comment above?

**Review by iliapolo — APPROVED** [review]

**Review by GavinZZ — APPROVED** [review]

**Comment by xazhao** [comment]
Synced with @iliapolo offline. Will remove the integration tests from the repo and add a few TODOs to the code

**Pull request merged**